python/neutron-understack/neutron_understack/neutron_understack_mech.py

```diff
@@ -204,8 +249,13 @@ def update_port_precommit(self, context):
     def update_port_postcommit(self, context):
         log_call("update_port_postcommit", context)
 
-        if context.current["binding:vif_type"] == portbindings.VIF_TYPE_OTHER:
-            self._move_to_network(context)
+        _move_to_network(
```

Comment:
I feel like this method should live in the  `UnderstackDriver` class rather than top-level scope - was there a reason why it was moved?

FWIW, same goes for `ArgoClient` - reading secrets for argo should also be part of the `ArgoClient` class, but that one is my fault as I wrote it in the wrong place.

Reply:
OK, moved the secret-reading into the Argo class in ddef1ed.

I only changed _move_to_network to make it more straightforward to test without getting tangled up in MechanismDriver.  However, it seemed like the UnderstackDriver class was already doing quite a lot (it implements the MechanismDriver interface, and has knowledge of the `context` structure).  "Move to network" seemed orthogonal, and wasn't using `self` for anything, so I dropped it out of the class.  I don't know how pythonic this is - I guess it is more normal to have every function in a class?

I moved those two module-level functions into the class in 42279a6 and 3fbcfa2.

Reply:
>  I don't know how pythonic this is

I think it's viable, but much harder to mock if needed in future